contrib/labstack/echo.v4/appsec.go

```diff
@@ -6,33 +6,52 @@
 package echo
 
 import (
+	"net/http"
+
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
-	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo/instrumentation"
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo/instrumentation/httpsec"
 
 	"github.com/labstack/echo/v4"
 )
 
-func useAppSec(c echo.Context, span tracer.Span) func() {
-	instrumentation.SetAppSecEnabledTags(span)
-
+func withAppSec(next echo.HandlerFunc, c echo.Context, span tracer.Span) echo.HandlerFunc {
 	params := make(map[string]string)
 	for _, n := range c.ParamNames() {
 		params[n] = c.Param(n)
 	}
+	return func(c echo.Context) error {
+		var err error
+		handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			err = next(c)
+			if err != nil {
+				c.Error(err)
```

Comment:
I don't think you need this: the request handler `next` or the calling middleware will do it. (to be confirmed checking c.Error's documentation)